daemon/inspect.go

```diff
@@ -134,6 +134,24 @@ func (daemon *Daemon) getInspectData(container *Container, size bool) (*types.Co
 		HostConfig:   &hostConfig,
 	}
 
+	// Make sure the dns fields are never nil.
+	// New containers don't ever have those fields nil,
+	// but pre created containers can still have those nil values.
+	// See https://github.com/docker/docker/pull/17779
+	// for a more detailed explanation on why we don't want that.
+	var emptySlice []string
```

Comment:
Won't this slice be `nil` unless you allocate it with `make` or by setting it to a slice literal?